### PR TITLE
58963 fix hgfs bytes vs strings issues

### LINKED ISSUE #58963: [BUG] hgfs fileserver backend still broken
**Description**
Issue https://github.com/saltstack/salt/issues/58191 hasn't been fixed. hgfs fileserver still doesn't work even after applying https://github.com/saltstack/salt/pull/58852

**Setup**
```
fileserver_backend:
  - hgfs

hgfs_remotes:
  - http://repo.com/my_repo  # replace by a valid repo url
```

**Steps to Reproduce the behavior**
`salt-run fileserver.file_list base hgfs` always shows nothing

**Expected behavior**
Get files from hgfs fileserver backend

**Versions Report**
current master 

**Research**
As long as I can find, the main reason is that `hglib` returns binary strings and salt expects standard python3 strings. In search for branch there is comparison between binary string returned from hglib (for example b'default') and target branch from saltenv (for example 'default'). This comparison is always False. So `_get_branch` function in `_get_ref` can't find branches.

### PR DESCRIPTION
**Pull request by whytewolf**
### What does this PR do?
This PR fixes hgfs 

### What issues does this PR fix or reference?
Fixes: #58963

### Previous Behavior
hgfs breaking do to bytes vs string issues in many places
### New Behavior
fixes bytes vs strings as well as a lot of functional testing that was being missed before. 

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
No

### CODE REVIEW

File: requirements/static/ci/linux.in [hunk 1]
```diff
@@ -9,3 +9,5 @@ ansible
 twilio
 python-telegram-bot==12.8; python_version <= '3.5'
 python-telegram-bot>=13.7; python_version > '3.5'
+hglib
```
Comment (Ch3LL) [inline requirements/static/ci/linux.in:12]: Are you certain hglib and mercurial only run on linux?

**Review by Ch3LL — COMMENTED** [review]
I just have one question

**Comment by whytewolf** [comment]
hglib and mercurial do not only run on linux. was mostly to avoid having to fight with pathing between windows, linux and other unixes. I can update it to run everywhere but will take more time to get the pathing right.

File: requirements/static/ci/linux.in (hunk 1 above)
Comment (whytewolf) [reply requirements/static/ci/linux.in]: alright, found out installing mercurial into windows is a bad idea. broke python pretty bad. and was breaking i before any tests ran. set up the tests to run on the rest of the operating systems. will have to skip windows.

**Review by garethgreenaway — APPROVED** [review]

**Pull request merged**